sklearn/covariance/graph_lasso_.py

```diff
@@ -191,6 +191,7 @@ def graph_lasso(emp_cov, alpha, cov_init=None, mode='cd', tol=1e-4,
     else:
         errors = dict(invalid='raise')
     try:
+        d_gap = np.inf
```

Comment:
Let's put an inline comment to make it explicit that this is to support the `max_iter=0` case.

Alternatively I would be ok to add input validation to ensure that models  configured with `max_iter < 1` yield consistently a `ValueError`.

I don't see the point in supporting the `max_iter=0` edge case in the first place. If most sklearn models already do something reasonable (a.k.a. then do not crash neither at `fit` nor `predict` or `transform` time) then this is is fine. But if this is problematic for any of them I would just disable explicitly support for `max_iter=0` rather than hacking workarounds to add support for this useless edge case.